### PR TITLE
[SPARK-16748][SQL] SparkExceptions during planning should not wrapped in TreeNodeException

### PR DESCRIPTION
**Pull request by tdas**
## What changes were proposed in this pull request?

We do not want SparkExceptions from job failures in the planning phase to create TreeNodeException. Hence do not wrap SparkException in TreeNodeException.
## How was this patch tested?

New unit test

### CODE REVIEW

**Comment by tdas** [comment]
@marmbrus

File: sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala [hunk 1]
```diff
@@ -20,7 +20,7 @@ package org.apache.spark.sql
 import java.math.MathContext
 import java.sql.Timestamp
 
-import org.apache.spark.AccumulatorSuite
+import org.apache.spark.{SparkException, AccumulatorSuite}
```
Comment (yhuai) [inline sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala]: import org.apache.spark.{AccumulatorSuite, SparkException}

**Comment by tdas** [comment]
jenkins, test this.

**Comment by yhuai** [comment]
seems jenkins is down?

**Comment by yhuai** [comment]
this this please

**Comment by yhuai** [comment]
LGTM. Merging to master and branch 2.0.

**Pull request closed without merging**